src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -832,11 +851,21 @@ ErrorIfInsertPartitionColumnDoesNotMatchSelect(Query *query, RangeTblEntry *inse
 				break;
 			}
 
+			cteListContext = list_make1(query->cteList);
+			originalRelationId = FindOriginalRelationId(subqeryTargetEntry->expr,
```

Comment:
I'm a bit confused why we now need this - especially the CTE bits seem entirely unrelated to view support?